### ISSUE TITLE
Speed and multiple GPU

### ISSUE THREAD

**Issue by tampe125** [KaiyangZhou/deep-person-reid#395]
Hello,  
this is more a question than a bug report.  

I'm currently working on dataset that contains more than 3000 unique identities, with a total of around 40000 images (I know, I should have more samples but this is what I have at hand...).  

Training with 20 epochs on my old GTX 970 takes about 2 hours; I can't complain since it's a pretty old card. For curiosity sake, I started spinning some AWS instances to see if I can speed things up. No matter which instance I pick (one old GPU, multiple old GPUs, one recent GPU, multiple recent GPUs), training time remains the same, or even worse (!!!).  

The output is rather interesting, too. I can see that the average for GPUs processing and data access is pretty high, but it's caused by "spikes" during the training, below you can see an output example:

```
epoch: [5/20][140/1198] time 0.046 (0.148)      data 0.030 (0.090)      eta 1:47:04     loss 1.4935 (1.5356)    acc 100.0000 (98.7723)  lr 0.000300
epoch: [5/20][150/1198] time 0.247 (0.148)      data 0.060 (0.090)      eta 1:47:01     loss 1.5491 (1.5389)    acc 96.8750 (98.7292)   lr 0.000300

```

Is there any settings to unlock the usage of multiple GPUs? If that's not possible or it's simply too complicated there's no problem, it was just a curiosity.

**Comment by jasonbrancazio** [comment]
@tampe125 did you ever figure out how to get training running with multiple GPUs? I see you self-closed the issue and @KaiyangZhou never responded.

I recently tested torchreid on Google Cloud's Vertex AI with a custom dataset. The run with 4 Tesla T4s was 50% **slower** than with 1 T4, and only 25% of the GPU pool was being utilized.

I tried 3 different ways to get multi-GPU training going:
1) doing nothing other than attaching more GPUs to the training job
2) explicitly setting env variable CUDA_VISIBLE_DEVICES to 0,1,2,3
3) launching the script with torch.distributed.launch

None worked. For #3, 4 copies of the training script were launched on separate workers, effectively just training and saving the model 4 times, 1 for each GPU.
